src/main/java/green/rijks/RijksFrame.java

```diff
@@ -0,0 +1,143 @@
+import hu.akarnokd.rxjava3.swing.SwingSchedulers;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.io.IOException;
+import java.net.URL;
+
+public class RijksFrame extends JFrame  {
+
+    private ApiKey key = new ApiKey();
+    private final RijksService service = new RijksServiceFactory().getService();
+    private int currPage = 1;
+    private ArtObjects artObjs;
+    private JPanel paintingComponent = new JPanel();
+
+    public RijksFrame() {
+        setSize(800, 600);
+        setTitle("Rijks Museum Paintings");
+        setDefaultCloseOperation(EXIT_ON_CLOSE);
+
+        JPanel main = new JPanel();
+        main.setLayout(new BorderLayout());
+        setContentPane(main);
+
+        JButton next = new JButton("Next");
+        JButton prev = new JButton("Previous");
+
+        JPanel prevPanel = new JPanel(new FlowLayout(FlowLayout.LEFT));
+        JPanel nextPanel = new JPanel(new FlowLayout(FlowLayout.RIGHT));
+        prevPanel.add(prev);
+        nextPanel.add(next);
+
+        main.add(prevPanel, BorderLayout.WEST);
+        main.add(nextPanel, BorderLayout.EAST);
+
+        JTextField search = new JTextField();
+        main.add(search, BorderLayout.PAGE_START);
+
+        main.add(paintingComponent, BorderLayout.CENTER);
+
+        onStart();
+
+        next.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                Disposable disposable = service.getPage(key.get(), ++currPage)
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(SwingSchedulers.edt())
+                        .subscribe(
+                                (response) -> handleResponse(response),
+                                Throwable::printStackTrace);
+            }
+        });
+
+        prev.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                Disposable disposable = service.getPage(key.get(), --currPage)
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(SwingSchedulers.edt())
+                        .subscribe(
+                                (response) -> handleResponse(response),
+                                Throwable::printStackTrace);
+            }
+        });
+
+
+        DocumentListener docListener = new DocumentListener() {
+            @Override
+            public void insertUpdate(DocumentEvent e) {
+                Disposable disposable = service.getField(key.get(), search.getText(), --currPage)
```

Comment:
are you sure you want to do `--currPage`?

If you do an artist search and you hit "Next Page", it's not going to give you the next page of that artist.